data-access/nexustiles/nexustiles.py

```diff
@@ -163,6 +163,20 @@ def find_tiles_in_polygon(self, bounding_polygon, ds=None, start_time=0, end_tim
                                                                      **kwargs)
         return tiles
 
+    @tile_data()
+    def find_tiles_by_metadata(self, metadata, ds=None, start_time=0, end_time=-1, **kwargs):
```

Comment:
I think this needs to apply a mask for start/end time (and in fact we will probably need to do this for most other methods in nexustileservice) since we can no longer assume that the tile will only contain data relevant to the requested start/end time. A tile might now partially overlap the requested time frame, so we should mask the irrelevant data.

Reply:
Good point. Since this affect other methods, I've created a new ticket https://issues.apache.org/jira/browse/SDAP-41 to track the changes to handle the temporal masking.